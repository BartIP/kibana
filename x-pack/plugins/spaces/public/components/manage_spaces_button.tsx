/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { EuiButton } from '@elastic/eui';
<<<<<<< HEAD
=======
import { FormattedMessage } from '@kbn/i18n/react';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import React, { Component, CSSProperties } from 'react';
import { UserProfile } from '../../../xpack_main/public/services/user_profile';
import { MANAGE_SPACES_URL } from '../lib/constants';

interface Props {
  isDisabled?: boolean;
  className?: string;
  size?: 's' | 'l';
  style?: CSSProperties;
  userProfile: UserProfile;
  onClick?: () => void;
}

export class ManageSpacesButton extends Component<Props, {}> {
  public render() {
    if (!this.props.userProfile.hasCapability('manageSpaces')) {
      return null;
    }

    return (
      <EuiButton
        size={this.props.size || 's'}
        className={this.props.className}
        isDisabled={this.props.isDisabled}
        onClick={this.navigateToManageSpaces}
        style={this.props.style}
      >
<<<<<<< HEAD
        Manage spaces
=======
        <FormattedMessage
          id="xpack.spaces.manageSpacesButton.manageSpacesButtonLabel"
          defaultMessage="Manage spaces"
        />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      </EuiButton>
    );
  }

  private navigateToManageSpaces = () => {
    if (this.props.onClick) {
      this.props.onClick();
    }
    window.location.replace(MANAGE_SPACES_URL);
  };
}
