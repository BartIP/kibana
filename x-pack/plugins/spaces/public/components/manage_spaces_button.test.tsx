/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
<<<<<<< HEAD
import { shallow } from 'enzyme';
import React from 'react';
=======
import React from 'react';
import { shallowWithIntl } from 'test_utils/enzyme_helpers';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { UserProfileProvider } from '../../../xpack_main/public/services/user_profile';
import { ManageSpacesButton } from './manage_spaces_button';

const buildUserProfile = (canManageSpaces: boolean) => {
  return UserProfileProvider({ manageSpaces: canManageSpaces });
};

describe('ManageSpacesButton', () => {
  it('renders as expected', () => {
    const component = <ManageSpacesButton userProfile={buildUserProfile(true)} />;
<<<<<<< HEAD
    expect(shallow(component)).toMatchSnapshot();
=======
    expect(shallowWithIntl(component)).toMatchSnapshot();
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  });

  it(`doesn't render if user profile forbids managing spaces`, () => {
    const component = <ManageSpacesButton userProfile={buildUserProfile(false)} />;
<<<<<<< HEAD
    expect(shallow(component)).toMatchSnapshot();
=======
    expect(shallowWithIntl(component)).toMatchSnapshot();
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  });
});
