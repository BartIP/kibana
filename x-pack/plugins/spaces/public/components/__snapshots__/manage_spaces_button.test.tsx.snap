// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageSpacesButton doesn't render if user profile forbids managing spaces 1`] = `""`;

exports[`ManageSpacesButton renders as expected 1`] = `
<EuiButton
  color="primary"
  fill={false}
  iconSide="left"
  onClick={[Function]}
  size="s"
  type="button"
>
<<<<<<< HEAD
  Manage spaces
=======
  <FormattedMessage
    defaultMessage="Manage spaces"
    id="xpack.spaces.manageSpacesButton.manageSpacesButtonLabel"
    values={Object {}}
  />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
</EuiButton>
`;
